client/components/checklist/index.js

```diff
@@ -16,37 +16,53 @@ import TaskPlaceholder from 'components/checklist/task-placeholder';
 export default class Checklist extends PureComponent {
 	static propTypes = {
 		isPlaceholder: PropTypes.bool,
+		updateCompletion: PropTypes.func,
 	};
 
-	state = { hideCompleted: false };
-
-	toggleCompleted = () =>
-		this.setState( ( { hideCompleted } ) => ( { hideCompleted: ! hideCompleted } ) );
+	componentDidMount() {
+		this.notifyCompletion();
+	}
 
-	renderPlaceholder() {
-		return (
-			<div className={ classNames( 'checklist', 'is-expanded', 'is-placeholder' ) }>
-				<ChecklistHeader total={ 0 } completed={ 0 } />
-				<div className="checklist__tasks">
-					{ times( Children.count( this.props.children ), index => (
-						<TaskPlaceholder key={ index } />
-					) ) }
-				</div>
-			</div>
-		);
+	componentDidUpdate() {
+		this.notifyCompletion();
 	}
 
-	render() {
-		if ( this.props.isPlaceholder ) {
-			return this.renderPlaceholder();
+	notifyCompletion() {
+		if ( 'function' === typeof this.props.updateCompletion ) {
+			const [ complete, total ] = this.calculateCompletion();
+			this.props.updateCompletion( { complete: complete >= total } );
 		}
+	}
 
+	calculateCompletion() {
 		const { children } = this.props;
-
-		const count = Children.map( children, child => child.props.completed ).reduce(
-			( acc, completed ) => ( true === completed ? acc + 1 : acc ),
+		const completedCount = Children.toArray( children ).reduce(
+			( count, task ) => ( true === task.props.completed ? count + 1 : count ),
 			0
 		);
+		const total = Children.count( children );
+		return [ completedCount, total ];
+	}
+
+	state = { hideCompleted: false };
+
+	toggleCompleted = () =>
+		this.setState( ( { hideCompleted } ) => ( { hideCompleted: ! hideCompleted } ) );
+
+	render() {
+		const [ completed, total ] = this.calculateCompletion();
+		if ( this.props.isPlaceholder ) {
+			return (
+				<div className={ classNames( 'checklist', 'is-expanded', 'is-placeholder' ) }>
+					<ChecklistHeader completed={ completed } total={ total } />
+					<div className="checklist__tasks">
+						{ times( Children.count( this.props.children ), index => (
```

Comment:
Could use `total` instead of `Children.count( this.props.children )` here, no?

Reply:
36d03d7b0daf74d16065786061a4856b96e22231